File: core/src/main/java/com/jinkyumpark/bookitout/search/service/SearchBookService.java
package com.jinkyumpark.bookitout.search.service;

import com.jinkyumpark.bookitout.search.apiResponse.aladin.ApiAladinItem;
import com.jinkyumpark.bookitout.search.apiResponse.aladin.ApiAladinResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.core.env.Environment;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

import java.util.List;

@RequiredArgsConstructor
@Service
public class SearchBookService {
    private final RestTemplate restTemplate;
    private final Environment environment;

    public List<ApiAladinItem> getAladinItemByQuery(String query, int size) {
        String requestUrl = String.format("http://www.aladin.co.kr/ttb/api/ItemSearch.aspx?TTBKey=%s&Query=%s&Output=JS&version=20131101&MaxResults=%s",
                environment.getProperty("search.aladin.api-key"), query, size);

        ApiAladinResponse response = restTemplate.getForObject(requestUrl, ApiAladinResponse.class);

        if (response == null) return List.of();

        return response.getItem();
    }
}

File: search/src/main/kotlin/com/jinkyumpark/search/bookSearch/SearchBookService.kt
package com.jinkyumpark.search.bookSearch

import com.jinkyumpark.search.apiResponse.aladin.ApiAladinResponse
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import org.springframework.web.reactive.function.client.WebClient

@Service
class SearchBookService(
    val webClient: WebClient,

    @Value("\${search.aladin.api-key}")
    val aladinApiKey: String
) {

    fun getBookByQueryFromAladin(query: String, size: Int): List<BookSearchResult> {
        val url = "http://www.aladin.co.kr/ttb/api/ItemSearch.aspx?TTBKey=$aladinApiKey&Query=$query&Output=JS&version=20131101&MaxResults=$size"

        val response: ApiAladinResponse = webClient
            .get()
            .uri(url)
            .retrieve()
            .bodyToMono(ApiAladinResponse::class.java)
            .block() ?: return listOf()

        return response.item?.map { aladin ->
            BookSearchResult(
                title = aladin.title ?: "",
                author = aladin.author,
                link = aladin.link,
                cover = aladin.cover,
                isbn = aladin.isbn13
            )
        } ?: listOf()
    }

}